output/transform/merge_partition_sizes.go

```diff
@@ -12,11 +12,21 @@ import (
 //
 // TODO: recursively build up stats when nested partitioning is used
 func mergePartitionSizes(s snapshot.FullSnapshot, newState state.PersistedState, ts state.TransientState, databaseOidToIdx OidToIdx) snapshot.FullSnapshot {
+	relIdxToStatsIdx := make(map[int32]int, len(s.RelationStatistics))
+	for idx, stat := range s.RelationStatistics {
+		relIdxToStatsIdx[stat.RelationIdx] = idx
+	}
+
 	for idx, rel := range s.RelationInformations {
 		if !rel.HasParentRelation || rel.PartitionBoundary == "" {
 			continue
 		}
-		stat := s.RelationStatistics[idx]
+		statIdx, ok := relIdxToStatsIdx[int32(idx)]
+		if !ok {
+			continue
+		}
+
+		stat := s.RelationStatistics[statIdx]
 		parent := s.RelationStatistics[rel.ParentRelationIdx]
```

Comment:
I think this line is also incorrect - we need to do the same lookup here as well.

Reply:
And we have the same problem for index statistics as well further down (i.e. we'd also need to build a map for index relation index to index stats entry index)